python/thunder/rdds/fileio/readers.py

```diff
@@ -438,23 +432,16 @@ def __getMatchingKeys(self, dataPath, filename=None):
                         keyName = ""
             keyName += filename
 
-        return _BotoS3Client.retrieveKeys(bucket, keyName, prefix=parse[2], postfix=parse[3])
+        return _BotoS3Client.retrieveKeys(bucket, keyName, prefix=parse[2], postfix=parse[3],
+                                          excludeDirectories=excludeDirectories, recursive=recursive)
 
-    def list(self, dataPath, filename=None, startIdx=None, stopIdx=None, recursive=False, includeDirectories=True):
+    def list(self, dataPath, filename=None, startIdx=None, stopIdx=None, recursive=False, includeDirectories=False):
         """List s3 objects specified by dataPath.
 
         Returns sorted list of 's3n://' URIs.
         """
-        # TODO: note that the default value for includeDirectories is here True, which reflects the current (and
-        # longstanding) behavior of this class. This *differs* from the local FS list() method, which by default
-        # filters out directories (which is definitely what we want if we're reading with recursive=True).
-        # These two need to be made more consistent.
-        if recursive:
-            raise NotImplementedError("Recursive traversal of directories isn't yet implemented for S3 - sorry!")
-        if not includeDirectories:
-            raise NotImplementedError("Filtering out directories is not yet implemented for S3 - sorry!")
-
-        keys = self.__getMatchingKeys(dataPath, filename=filename)
+        keys = self.__getMatchingKeys(dataPath, filename=filename, excludeDirectories=not includeDirectories,
```

Comment:
Having a little trouble following the double negative / use of include vs exclude, it didn't make sense to just use one or the other throughout?